owid/catalog/datasets.py

```diff
@@ -169,29 +190,22 @@ def _index_file(self) -> str:
         return join(self.path, "index.json")
 
     def __len__(self) -> int:
-        return len(self._data_files)
+        return len(self.table_names)
 
     def __iter__(self) -> Iterator[tables.Table]:
-        for filename in self._data_files:
-            if filename.endswith(".feather"):
-                yield tables.Table.read_feather(filename)
-
-            elif filename.endswith(".csv"):
-                yield tables.Table.read_csv(filename)
-
-            else:
-                raise Exception(f"don't know how to read table: {filename}")
+        for name in self.table_names:
+            yield self[name]
 
     @property
     def _data_files(self) -> List[str]:
         feather_pattern = join(self.path, "*.feather")
+        parquet_pattern = join(self.path, "*.feather")
```

Comment:
@larsyencken should this be `*.feather`?

Reply:
Yes! Hotfixed now.